uploadDeployGate task fails if `skipAssemble = true` is set without `sourceFile`

### Summary

`uploadDeployGate<capitalized VariantName>` task fails with following message if `skipAssemble = true` is set and `sourceFile` is not set in configuration.

```
* What went wrong:
Execution failed for task ':app:uploadDeployGateDebug'.
> An apk file to be upload not specified.
```

### How to reproduce

I created a sample app project to reproduce the issue.
https://github.com/kafumi/deploygate-plugin-upload-issue

Here is example output when I execute `uploadDeployGateDebug` task following `assembleDebug` task.

```
$ ./gradlew assembleDebug

BUILD SUCCESSFUL in 4s
26 actionable tasks: 26 executed

$ ./gradlew uploadDeployGateDebug
> Task :app:uploadDeployGateDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:uploadDeployGateDebug'.
> An apk file to be upload not specified.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
2 actionable tasks: 2 executed
```

Here is the `deploygate` configuraion in `app/build.gradle`.

```groovy
deploygate {
    deployments {
        debug {
            skipAssemble = true
        }
    }
}
```

### Workarounds

- The issue doesn't happen if I set explicitly set `sourceFile` in configuration
  - Sample branch I used to check this workaround: https://github.com/kafumi/deploygate-plugin-upload-issue/tree/explicit-source-file
- The issue doesn't happen if `package<Variant>` task is executed along with `uploadDeployGate<Variant>` task
```
$ ./gradlew assembleDebug

BUILD SUCCESSFUL in 2s
26 actionable tasks: 26 executed

$ ./gradlew packageDebug uploadDeployGateDebug

BUILD SUCCESSFUL in 27s
28 actionable tasks: 2 executed, 26 up-to-date
```

### Notes

- The issue doesn't heppen if I use Gradle DeployGate plugin v1
  - Sample branch with v1 plugin: https://github.com/kafumi/deploygate-plugin-upload-issue/tree/with-v1-plugin

--- a/src/main/groovy/com/deploygate/gradle/plugins/tasks/factory/AGPBasedUploadApkTaskFactory.groovy
+++ b/src/main/groovy/com/deploygate/gradle/plugins/tasks/factory/AGPBasedUploadApkTaskFactory.groovy
@@ -39,6 +39,12 @@ class AGPBasedUploadApkTaskFactory extends DeployGateTaskFactory implements Uplo
             dgTask.configuration = configuration
             dgTask.applyTaskProfile()
         }
+
+        if (deployment?.skipAssemble) {
+            // a package application provider may not be evaluated in some cases.
+            // ref: https://github.com/DeployGate/gradle-deploygate-plugin/issues/86
+            applicationVariant.lazyPackageApplication().get()
+        }
     }
 
     @Override

--- a/src/test/acceptance/com/deploygate/gradle/plugins/AcceptanceKtsTestSpec.groovy
+++ b/src/test/acceptance/com/deploygate/gradle/plugins/AcceptanceKtsTestSpec.groovy
@@ -14,7 +14,8 @@ class AcceptanceKtsTestSpec extends AcceptanceTestBaseSpec {
     AGPEnv[] getTestTargetAGPEnvs() {
         return [
                 new AGPEnv("3.3.2", "4.10.1"),
-                new AGPEnv("3.4.0-beta05", "5.1.1"),
+                new AGPEnv("3.4.0", "5.1.1"),
+                new AGPEnv("3.5.0", "5.4.1"),
         ]
     }
 

--- a/src/test/acceptance/com/deploygate/gradle/plugins/AcceptanceTestBaseSpec.groovy
+++ b/src/test/acceptance/com/deploygate/gradle/plugins/AcceptanceTestBaseSpec.groovy
@@ -261,26 +261,35 @@ abstract class AcceptanceTestBaseSpec extends Specification {
     }
 
     @Unroll
-    def "flavor2Flavor4Debug needs assembling #agpVersion"() {
+    def "flavor2Flavor4Debug require assembling #agpVersion"() {
         given:
         testAndroidProject.gradleProperties([
                 "agpVersion": agpVersion
         ])
 
-        def runner = GradleRunner.create()
+        def assembleRunner = GradleRunner.create()
                 .withProjectDir(testProjectDir.root)
                 .withPluginClasspath(testDeployGatePlugin.loadPluginClasspath())
                 .withGradleVersion(gradleVersion)
-                .withArguments("assembleFlavor2Flavor4Debug", "uploadDeployGateFlavor2Flavor4Debug" /*, "--stacktrace" */)
+                .withArguments("assembleFlavor2Flavor4Debug" /*, "--stacktrace" */)
+
+        def uploadRunner = GradleRunner.create()
+                .withProjectDir(testProjectDir.root)
+                .withPluginClasspath(testDeployGatePlugin.loadPluginClasspath())
+                .withGradleVersion(gradleVersion)
+                .withArguments("uploadDeployGateFlavor2Flavor4Debug" /*, "--stacktrace" */)
 
         and:
-        def buildResult = runner.build()
+        def assembleBuildResult = assembleRunner.build()
+
+        and:
+        def uploadBuildResult = uploadRunner.build()
         def result = wireMockRule.findRequestsMatching(postRequestedFor(urlPathEqualTo("/api/users/appOwner/apps")).build())
         def request = result.requests.first()
 
         expect:
-        buildResult.task(":assembleFlavor2Flavor4Debug").outcome == TaskOutcome.SUCCESS
-        buildResult.task(":uploadDeployGateFlavor2Flavor4Debug").outcome == TaskOutcome.SUCCESS
+        assembleBuildResult.task(":assembleFlavor2Flavor4Debug").outcome == TaskOutcome.SUCCESS
+        uploadBuildResult.task(":uploadDeployGateFlavor2Flavor4Debug").outcome == TaskOutcome.SUCCESS
         result.requests.size() == 1
         request.getPart("token").body.asString() == "api token"
         request.getPart("file").body.present

--- a/src/test/acceptance/com/deploygate/gradle/plugins/AcceptanceTestSpec.groovy
+++ b/src/test/acceptance/com/deploygate/gradle/plugins/AcceptanceTestSpec.groovy
@@ -14,7 +14,8 @@ class AcceptanceTestSpec extends AcceptanceTestBaseSpec {
                 new AGPEnv("3.1.0", "4.4"),
                 new AGPEnv("3.2.0", "4.6"),
                 new AGPEnv("3.3.2", "4.10.1"),
-                new AGPEnv("3.4.0-beta05", "5.1.1"),
+                new AGPEnv("3.4.1", "5.1.1"),
+                new AGPEnv("3.5.0", "5.4.1"),
         ]
     }
 }

--- a/src/test/acceptance/com/deploygate/gradle/plugins/internal/agp/AndroidGradlePluginAcceptanceSpec.groovy
+++ b/src/test/acceptance/com/deploygate/gradle/plugins/internal/agp/AndroidGradlePluginAcceptanceSpec.groovy
@@ -63,6 +63,7 @@ class AndroidGradlePluginAcceptanceSpec extends Specification {
         "3.1.0"          | "4.4"
         "3.2.0"          | "4.6"
         "3.3.2"          | "4.10.1"
-        "3.4.0-beta05"   | "5.1.1"
+        "3.4.1"          | "5.1.1"
+        "3.5.0"          | "5.4.1"
     }
 }

--- a/src/test/acceptance/com/deploygate/gradle/plugins/internal/gradle/GradleCompatAcceptanceSpec.groovy
+++ b/src/test/acceptance/com/deploygate/gradle/plugins/internal/gradle/GradleCompatAcceptanceSpec.groovy
@@ -52,7 +52,8 @@ class GradleCompatAcceptanceSpec extends Specification {
                 "4.8",
                 "4.9",
                 "4.10.1",
-                "5.1.1"
+                "5.1.1",
+                "5.4.1"
         ]
     }
 }

